TombRaiderIII/Autosplitter.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using LiveSplit.ComponentUtil;
+using LiveSplit.Model;
+using LiveSplit.Options;
+using LiveSplit.UI.Components.AutoSplit;
+
+namespace TR3
+{
+    /// <summary>
+    ///     The game's level and demo values.
+    /// </summary>
+    [SuppressMessage("ReSharper", "UnusedMember.Global")]
+    internal enum Level : uint
+    {
+        LarasHome = 00,
+        // India
+        Jungle = 01,
+        TempleRuins = 02,
+        TheRiverGanges = 03,
+        CavesOfKaliya = 04,
+        // South Pacific
+        CoastalVillage = 05,
+        CrashSite = 06,
+        MadubuGorge = 07,
+        TempleOfPuna = 08,
+        // London
+        ThamesWharf = 09,
+        Aldwych = 10,
+        LudsGate = 11,
+        City = 12,
+        // Nevada
+        NevadaDesert = 13,
+        HighSecurityCompound = 14,
+        Area51 = 15,
+        // Antarctica
+        Antarctica = 16,
+        RxTechMines = 17,
+        LostCityOfTinnos = 18,
+        MeteoriteCavern = 19,
+        // Bonus (All Secrets)
+        AllHallows = 20
+    }
+
+    /// <summary>
+    ///     Implementation of <see cref="IAutoSplitter"/> for an <see cref="AutoSplitComponent"/>'s use.
+    /// </summary>
+    internal class Autosplitter : IAutoSplitter, IDisposable
+    {
+        private const int IgtTicksPerSecond = 30;
+        private const int LevelCount = 20; // This considers Lara's Home as 0 and includes the bonus level.
+
+        private readonly HashSet<Level> _completedLevels = new HashSet<Level>(LevelCount);
+        private Level _startingLevel;
+
+        internal readonly ComponentSettings Settings = new ComponentSettings();
+        internal GameMemory GameMemory = new GameMemory();
+
+        /// <summary>A constructor that primarily exists to handle events/delegations.</summary>
+        public Autosplitter() => GameMemory.OnGameFound += Settings.SetGameVersion;
+
+        /// <summary>
+        ///     Determines the IGT.
+        /// </summary>
+        /// <param name="state"><see cref="LiveSplitState"/> passed by LiveSplit</param>
+        /// <returns>IGT as a <see cref="TimeSpan"/> if available, otherwise <see langword="null"/></returns>
+        public TimeSpan? GetGameTime(LiveSplitState state)
+        {
+            uint currentLevelTicks = GameMemory.Data.LevelTime.Current;
+            double currentLevelTime = (double)currentLevelTicks / IgtTicksPerSecond;
+
```

Comment:
I think the rest of this function could be simply `return TimeSpan.FromSeconds(currentLevelTime + SumCompletedLevelTimes());`, since `SumCompletedLevelTimes` returns 0 if there are no completed levels.